Makefile

```diff
@@ -132,8 +132,10 @@ else ifeq ($(MM_NO_DOCKER),true)
 else
 	@echo Starting docker containers
 
-	docker-compose run --rm start_dependencies
-	cat tests/${LDAP_DATA}-data.ldif | docker-compose exec -T openldap bash -c 'ldapadd -x -D "cn=admin,dc=mm,dc=test,dc=com" -w mostest || true';
+	$(GO) run ./build/docker-compose-generator/main.go $(ENABLED_DOCKER_SERVICES) | docker-compose -f docker-compose.makefile.yml -f /dev/stdin run --rm start_dependencies
```

Comment:
The `-f /dev/stdin` forces me to run through the `Makefile` in all cases, and I can't (easily) use `docker-compose` directly. 

I note from https://docs.docker.com/compose/extends/#multiple-compose-files that 
> By default, Compose reads two files, a `docker-compose.yml` and an optional `docker-compose.override.yml` file. By convention, the `docker-compose.yml` contains your base configuration. The override file, as its name implies, can contain configuration overrides for existing services or entirely new services.

Could we instead have add `docker-compose.override.yml` to `.gitignore` and unconditionally generate it on each `Makefile` invocation? Then it's around and ready for someone to use with `docker-compose` directly.

Reply:
Actually, we have 2 docker-compose files, one is the `docker-compose.makefile.yml` that is the one used from the makefile, and it is exactly the same as `docker-compose.yaml` but without the autostart block, the autostart block is added dynamically by the `./build/docker-compose-generator/main.go`.

The `docker-compose.yaml` is reserved for the regular docker-compose usage, without nothing to do with our own `docker-compose.makefile.yml` that you can use if you want to start the containers individually. I have tried to be conservative and not break anybodies workflow here.

Reply:
Ah, excellent! I'm worried about the duplication here -- these will invariably drift out of sync when only testing one workflow. Can we generate the `docker-compose.makefile.yml` on the go instead?

Reply:
We can generate the whole thing on the fly, I would add that as another PR, I don't think is strictly neede for this PR.

Reply:
Any reason not to "get this right" with the first pass? I'm pushing back slightly because this affects every single developer using Mattermost and immediately hits a wide audience.